### ISSUE TITLE
CssSyntaxError breaks stylelint

### ISSUE THREAD

**Issue by wroughtec** [stylelint/stylelint#1161]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Bug when SASS variables are used inside #{$var} with a placeholder (maybe other locations as well).

> Which rule, if any, is this issue related to?

CssSyntaxError

> What CSS is needed to reproduce this issue?

```css
.class--discount {
  &::before {
    @extend %#{$fa-css-prefix}-check;
  }
}
```

> Which version of stylelint are you using?

6.2.2

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI, PostCSS plugin and via sublimetext

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes SCSS

> What did you expect to happen?

No warnings to be flagged.
Or at the very least not to break the rest of the file (did not used too in 5.4)

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
stylelint "./scss/_components/_c-main-nav.scss"
CssSyntaxError: /---/scss/_components/_c-main-nav.scss:117:18: Unknown word
    at Input.error (/usr/local/lib/node_modules/postcss/lib/input.js:61:22)
    at Parser.unknownWord (/usr/local/lib/node_modules/postcss/lib/parser.js:457:26)
    at Parser.word (/usr/local/lib/node_modules/postcss/lib/parser.js:174:14)
    at Parser.loop (/usr/local/lib/node_modules/postcss/lib/parser.js:60:26)
    at parse (/usr/local/lib/node_modules/postcss/lib/parse.js:26:12)
    at new LazyResult (/usr/local/lib/node_modules/postcss/lib/lazy-result.js:61:24)
    at Processor.process (/usr/local/lib/node_modules/postcss/lib/processor.js:34:16)
    at lintString (/usr/local/lib/node_modules/stylelint/dist/standalone.js:88:9)
    at /usr/local/lib/node_modules/stylelint/dist/standalone.js:70:14
```

**Comment by evilebottnawi** [comment]
@wroughtec run `stylelint --syntax scss ./scss/_components/_c-main-nav.scss`

**Comment by wroughtec** [comment]
👍 

Cool now just need to work out how I can pass this into sublime text

**Comment by evilebottnawi** [comment]
@wroughtec i don't know sublime, maybe settings exist for this options?

**Comment by wroughtec** [comment]
Can't see them have mentioned it on the plugin git. I would still say the error above should probably not break everything though should just report an error on that line.

**Comment by evilebottnawi** [comment]
@wroughtec http://sublimelinter.readthedocs.io/en/latest/linter_settings.html

**Comment by evilebottnawi** [comment]
@wroughtec see args

**Comment by wroughtec** [comment]
The args did not work it actually required some editing of the plugin to accept the args:
https://github.com/kungfusheep/SublimeLinter-contrib-stylelint/issues/15

That fix sublime for me thanks my red errors are back - never thought I would be happy to see errors lol

**Comment by evilebottnawi** [comment]
@wroughtec all work fine? close?
